sourcegen-model/src/main/java/io/micronaut/sourcegen/model/ExpressionDef.java

```diff
@@ -677,12 +688,98 @@ record NewArrayOfSize(TypeDef.Array type, int size) implements ExpressionDef {
     /**
      * The new array expression.
      *
-     * @param type       The type
+     * @param type        The type
      * @param expressions The items expression
      * @author Denis Stepanov
      * @since 1.2
      */
     @Experimental
-    record NewArrayInitialized(TypeDef.Array type, List<ExpressionDef> expressions) implements ExpressionDef {
+    record NewArrayInitialized(TypeDef.Array type,
+                               List<ExpressionDef> expressions) implements ExpressionDef {
+    }
+
+    /**
+     * The get property value expression.
+     *
+     * @param instance        The instance
+     * @param propertyElement The property element
+     * @author Denis Stepanov
+     * @since 1.3
+     */
+    @Experimental
+    record GetPropertyValue(ExpressionDef instance,
+                            PropertyElement propertyElement) implements ExpressionDef {
+
+        @Override
+        public TypeDef type() {
+            return TypeDef.of(propertyElement.getType());
+        }
+    }
+
+    /**
+     * The get class expression.
+     *
+     * @param instance The instance
+     * @author Denis Stepanov
+     * @since 1.3
+     */
+    @Experimental
+    record GetClassValue(ExpressionDef instance) implements ExpressionDef {
+
+        @Override
+        public TypeDef type() {
+            return TypeDef.of(Class.class);
+        }
+    }
+
+    /**
+     * The get hashCode expression.
+     *
+     * @param instance The instance
+     * @author Denis Stepanov
+     * @since 1.3
+     */
+    @Experimental
+    record HashCode(ExpressionDef instance) implements ExpressionDef {
```

Comment:
Can you add a functional API:
```java
EpxressionDef.HashCode hashCode() {
  return new HashCode(this);
}
```
I think we should mention that the functional API is preferred, as well:
```java
/**
     * The get hashCode expression. Where possible use {@link ExpressionDef#hashCode} method
     * to create a hash code for an expression.
     *
     * @param instance The instance
     * @author Denis Stepanov
     * @since 1.3
     */
```

Reply:
Added  `invokeHashCode` because we already have `invoke` for methods